src/shims/windows/foreign_items.rs

```diff
@@ -119,10 +122,59 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     system_info.ptr,
                     iter::repeat(0u8).take(system_info.layout.size.bytes() as usize),
                 )?;
+                // Set selected fields.
+                let word_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::U16));
+                let dword_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::U32));
+                let usize_ty = this.tcx.mk_ty(TyKind::Uint(UintTy::Usize));
+                let word_layout = this.layout_of(word_ty)?;
+                let dword_layout = this.layout_of(dword_ty)?;
+                let usize_layout = this.layout_of(usize_ty)?;
```

Comment:
You can use the fields of `this.machine.layouts`, no need to compute new layouts.

Reply:
Nice, I hoped they were already defined somewhere, but couldn't find them.

There is no `i16`/`u16` defined there, however. Should I just add them to `machine.rs`, or is it not worth it?

Reply:
I think better to add them in the long run.

Reply:
Yes please add them. :)